doc/Comparisons.rst

```diff
@@ -31,14 +17,40 @@ Panel and Dash can both be used to create dashboards in Python, but take very di
+
+The other major difference between Bokeh Server and Voila is the way they process notebook files. Voila is built directly on the notebook format. By default all output in the notebook is included in the rendered Voila app, which means existing notebooks can be served as apps _unchanged_. While that can be useful to get a quick set of plots, an existing notebook is unlikely to be organized and formatted in a way that forms a coherent dashboard, so in practice a notebook will need to be rewritten (some outputs suppressed, others rearranged) before it will make a good Voila dashboard, meaning that you end up with two copies of the notebook (one optimized to be a narrative, storytelling notebook with a series of cells, and another organized as a dashboard. Or you can write a template to select only the cells you want in the dashboard and rearrange them, but then you need to maintain both the notebook and the template separately.
+
+Panel takes a different approach in that output from a notebook cell needs to be explicitly wrapped in a Panel object and marked as being "servable"; cell outputs by default are shown only in the notebook, and not with ``panel serve``.  Panel in fact entirely ignores the fact that your notebook is organized into cells; it simply processes all the cells as Python code, and serves all the items that ended up being marked "servable".  Although this approach means editing the original notebook, it makes it practical for the same notebook to serve both an exploratory or storytelling purpose (in Jupyter) and act as a dashboard deployment (of a designated subset of the functionality). 
+
+
+Comparing Panel and streamlit
+=============================
+
+streamlit is an alternative to all of the above packages, including Jupyter, providing an interactive, incremental way to build apps (as in Jupyter), but by writing a Python file in a separate editor (as in Dash or Bokeh, and as opposed to using a cell-based Jupyter notebook). With streamlit, the entire source file is re-run every time a widget changes value, which has the advantage of not allowing confusing out-of-order execution of notebook cells. Using a Python source file allows users to choose their favorite editor rather than being limited to the web-based notebook interface, but it also typically means storing files locally rather than the client-server approach of notebooks that lets the source code live at a remote server while the user interface is local. In any case, for rerunning the entire file to be practical in any nontrivial case, streamlit requires all lengthy computations to be made cacheable, which is not always straightforward. A given Streamlit file produces a single application, as for Dash, though it can be incrementally built up in a simpler way than for Dash.
```

Comment:
It's true that it's not about Dash, but what I'm trying to convey here is that streamlit is meant to be a dynamically and continuously updated output from a given source file, with caching to make that feasible, while Dash is a more batch-oriented approach, without any special support for continuously updating the output efficiently while editing the file.   If there's no clear way to say that, then it can be deleted, but I think there's _something_ there to make clear...

Reply:
Dash supports hot reload (https://dash.plot.ly/devtools) to automatically update a dashboard on file save, but you're right that it doesn't have the kind of caching support that streamlit has.

Reply:
Ok, I'll update it.